chore: reorganize readme (#302)
timescale/pgai: README.md

<div align=center>

> <h3>pgai is your PostgreSQL-powered data layer for AI applications</h3>

Reviewer: This top line matters a lot. 

We could go a few ways:
- Build AI applications with PostgreSQL.
- Or describe pgai at a high level, like we do above.

Reviewer: ```suggestion
<h3> Power your AI applications with PostgreSQL
 </h3>
```

Reviewer 2: data layer is too vague imo I like the previous header but avthars suggestion works too.

[![Discord](https://img.shields.io/badge/Join_us_on_Discord-black?style=for-the-badge&logo=discord&logoColor=white)](https://discord.gg/KRdHVXAmkp)
[![Try Timescale for free](https://img.shields.io/badge/Try_Timescale_for_free-black?style=for-the-badge&logo=timescale&logoColor=white)](https://tsdb.co/gh-pgai-signup)
</div>

pgai is a PostgreSQL extension that simplifies data storage and retrieval for [Retrieval Augmented Generation](https://en.wikipedia.org/wiki/Prompt_engineering#Retrieval-augmented_generation) (RAG), and other AI applications.

> In particular, it automates the creation of embeddings for your data stored in PostgreSQL, simplifies

Reviewer: ```suggestion
In particular, it automates the creation and sync of embeddings for your data stored in PostgreSQL, simplifies
```

[semantic search](https://en.wikipedia.org/wiki/Semantic_search), and allows you to call LLM models from SQL.

<div align=center>

[![Auto Create and Sync Vector Embeddings in 1 Line of SQL (pgai Vectorizer)](https://github.com/user-attachments/assets/8a71c774-505a-4335-8b34-cdea9dedb558)](https://youtu.be/ZoC2XYol6Zk?si=atI4XPurEifG0pd5)

</div>

# Quick Start with Ollama

This section will walk you through the steps to get started with pgai and Ollama and show you the major features of pgai.

### Setup

> After checking out the repo, change directory to `examples/docker_compose_pgai_ollama` and run the following command to start the docker compose file:

Reviewer 2: I think allowing to copy the compose from the readme is less friction that cloning the whole repository. This e.g. allows people to create a compose in the project that they are currently working on which makes integrating easier later, since they can reuse the same file.

I get that you don't want the ~20 lines of compose here, but we can use a spoler/details tag for that like this:

<details>
<summary> docker-compose.yaml content:</summary>

```yaml
name: pgai
services:
  db:
    image: timescale/timescaledb-ha:pg17
    environment:
      POSTGRES_PASSWORD: postgres
    ports:
      - "5432:5432"
    volumes:
      - data:/home/postgres/pgdata/data
    command: [ "-c", "ai.ollama_host=http://ollama:11434" ]
  vectorizer-worker:
    image: timescale/pgai-vectorizer-worker:v0.3.0
    environment:
      PGAI_VECTORIZER_WORKER_DB_URL: postgres://postgres:postgres@db:5432/postgres
      OLLAMA_HOST: http://ollama:11434
    command: [ "--poll-interval", "5s", "--log-level", "DEBUG" ]
  ollama:
    image: ollama/ollama
volumes:
  data:
```
</details>

Author: I opted for a curl command to download instead. Tell me what you think.

Reviewer 2: I guess that's a decent middle ground :D

docker compose exec ollama ollama pull tinyllama
```

### Create a table and run a vectorizer

1. **Connect to the database in your local developer environment**
   The easiest way connect to the database is with the following command:
   `docker compose exec -it db psql`. 
   
   Alternatively, you can connect to the database with the following connection string: `postgres://postgres:postgres@localhost:5432/postgres`.

1. **Enable pgai on your database**

    ```sql
    CREATE EXTENSION IF NOT EXISTS ai CASCADE;
    ```
    
1. **Create a table with the data you want to embed from a huggingface dataset**

    We'll create a table named `wiki` from a few rows of the english-language `wikimedia/wikipedia` dataset.

    ```sql
    SELECT ai.load_dataset('wikimedia/wikipedia', '20231101.en', table_name=>'wiki', batch_size=>5, max_batches=>1);
    ```
    
    Related documentation: [load dataset from huggingface](/docs/load_dataset_from_huggingface.md).
    
    This table will contain the following columns: `id`, `url`, `title`, `text`. We'll create a primary key on the `id` column:

    ```sql
    ALTER TABLE wiki ADD PRIMARY KEY (id);

>     ```

Reviewer 2: I'd love to A/B test this. My gut feeling is that creating a sample table with standard postgres DDL and clear manual inserts highlights better how native this integrates into postgres.

Including the huggingface download here is also another level of magic and therefore perceived complexity. It's great for e.g. a video demo or blog post when someone is already involved but for trying to understand this project at a glance... I'm not convinced this is a good idea.

(again inserts could be hidden behind a `<details>` tag)
